packages/gatsby/cache-dir/dev-loader.js

```diff
@@ -2,9 +2,41 @@ import { BaseLoader, PageResourceStatus } from "./loader"
 import { findPath } from "./find-path"
 
 class DevLoader extends BaseLoader {
-  constructor(syncRequires, matchPaths) {
-    const loadComponent = chunkName =>
-      Promise.resolve(syncRequires.components[chunkName])
+  constructor(lazyRequires, matchPaths) {
+    // One of the tests doesn't set a path.
+    const loadComponent = chunkName => {
+      if (process.env.NODE_ENV !== `test`) {
+        delete require.cache[
+          require.resolve(`$virtual/lazy-client-sync-requires`)
+        ]
+        lazyRequires = require(`$virtual/lazy-client-sync-requires`)
+      }
+      if (lazyRequires.lazyComponents[chunkName]) {
+        return Promise.resolve(lazyRequires.lazyComponents[chunkName])
+      } else {
+        return new Promise(resolve => {
+          // Tell the server the user wants to visit this page
+          // to trigger it compiling the page component's code.
+          const req = new XMLHttpRequest()
+          req.open(`post`, `/___client-page-visited`, true)
+          req.setRequestHeader(`Content-Type`, `application/json;charset=UTF-8`)
+          req.send(JSON.stringify({ chunkName }))
+
+          const checkForUpdates = setInterval(() => {
+            if (process.env.NODE_ENV !== `test`) {
+              delete require.cache[
+                require.resolve(`$virtual/lazy-client-sync-requires`)
+              ]
+            }
+            const lazyRequires = require(`$virtual/lazy-client-sync-requires`)
+            if (lazyRequires.lazyComponents[chunkName]) {
+              clearInterval(checkForUpdates)
+              resolve(lazyRequires.lazyComponents[chunkName])
+            }
+          }, 100)
```

Comment:
setInterval can put lots of pressure on the event loop when the browser can't keep up. setTimeout relieves pressure.

```suggestion
          const checkForUpdates = () => {
            if (process.env.NODE_ENV !== `test`) {
              delete require.cache[
                require.resolve(`$virtual/lazy-client-sync-requires`)
              ]
            }
            const lazyRequires = require(`$virtual/lazy-client-sync-requires`)
            if (lazyRequires.lazyComponents[chunkName]) {
              return resolve(lazyRequires.lazyComponents[chunkName])
            }
            
            setTimeout(checkForUpdates, 100)
          }
          setTimeout(checkForUpdates, 100)
```

Reply:
👍